src/components/Main/Members/Members.tsx

```diff
@@ -0,0 +1,162 @@
+  const [isAddDialogOpen, setIsAddDialogOpen] = useState<boolean>(false);
+  const isSelected = (id: number) => selectedRows.indexOf(id) !== -1;
+
+  useEffect(() => {
+    dispatch(onLoadMembers());
+    dispatch(onLoadUser(initialSelectedState));
+  }, [])
+
+  const onOpenDeleteMemberDialog = () => {
+    if (selectedRows.length > 0) setIsConfirmDialogOpen(true);
+  };
+  
+  const onOpenAddMemberDialog = () => {
+    setIsAddDialogOpen(true);
+  }
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelectedRows = members.map((member) => member.id);
+      setSelectedRows(newSelectedRows);
+      return;
+    }
+    setSelectedRows([]);
+  }
+
+  const onCloseDeleteMemberDialog = async (value: boolean) => {
+    setIsConfirmDialogOpen(false);
+    if (value) {
+      dispatch(onDeleteMembers(selectedRows))
+      setSelectedRows([]);
+    }
+  };
+
+  const onCloseAddMemberDialog = (value: boolean, firstName: string, lastName: string, email: string, password: string) => {
+    setIsAddDialogOpen(false);
+    if (value && firstName && lastName && email && password && validateEmail(email)) {
+      dispatch(onAddMember(firstName, lastName, email, password));
+    }
+  }
+
+  const handleRowClick = (event: React.MouseEvent<unknown>, row: MemberStateData) => {
+    event.stopPropagation();
+    const selectedIndex = selectedRows.indexOf(row.id);
+    console.log(selectedRows, selectedIndex, selectedRows.slice(1));
+    let newSelectedRows: number[] = [];
+    if (event.ctrlKey) {
+      if (selectedIndex === -1) {
+        newSelectedRows = newSelectedRows.concat(selectedRows, row.id);
+      } else if (selectedIndex === 0) {
+        newSelectedRows = newSelectedRows.concat(selectedRows.slice(1));
+      } else if (selectedIndex === selectedRows.length -1) {
+        newSelectedRows = newSelectedRows.concat(selectedRows.slice(0, -1));
+      } else if (selectedIndex > 0) {
+        newSelectedRows = newSelectedRows.concat(
+          selectedRows.slice(0, selectedIndex), 
+          selectedRows.slice(selectedIndex + 1),
+        );
+      }
+    } else {
+      if (selectedIndex === -1)
+        newSelectedRows = [row.id]
+      else
+        newSelectedRows = newSelectedRows.concat(selectedRows.slice(0, selectedIndex), selectedRows.slice(selectedIndex + 1));
+    }
+    dispatch(onLoadUser(row));
+    setSelectedRows(newSelectedRows);
+  }
+
+  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    console.log(members)
+    setSearchField(event.target.value);
+  }
+
+  const filteredUsers = members.filter(function(row: any) {
+    for (var key in row) {
+      if (key === 'roles' || key === 'id' || key === 'ChurchId' || key === 'church') continue;
+      if (key === 'disabled') {
+        if (row[key].toString().toLowerCase().includes(searchfield.toLowerCase())) return true;
```

Comment:
@showduhtung this doesn't work --> that line ends up only filtering for firstName and everything else becomes ignored

Reply:
you don't have to use exactly what I wrote, just the principle of it. it's just that typically, if else statements are written out as `if (this) return true; else false`

my comment is just to rewrite to fulfill that concept

Reply:
the logic could simplify a lot, depending on this: why bother checking whether the user is disabled, if you're going to check the field either way?
```js
    for (let key in row) {
      const isUnsearchableKey = (key === 'roles' || key === 'id' || key === 'ChurchId' || key === 'church');
      if (isUnsearchableKey) continue;
      const foundMatch = row[key].toString().toLowerCase().includes(searchfield.toLowerCase());
      if (foundMatch) return true;
    }
    return false;
```

Reply:
@hchiam it's not checking if the user is disabled its just the variable type is different depending on the key, and I need to process them all as strings for the filtering

Reply:
@showduhtung I'm gonna leave this the way it is because that seems to be the only way it works, I'll leave it to you to make it work if you want to when I merge this branch